assets/js/blocks/cart-checkout/mini-cart-contents/inner-blocks/empty-mini-cart-contents-block/edit.tsx

```diff
@@ -1,40 +1,39 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
 import { useBlockProps, InnerBlocks } from '@wordpress/block-editor';
-import { innerBlockAreas } from '@woocommerce/blocks-checkout';
-import type { TemplateArray } from '@wordpress/blocks';
 import { useEditorContext } from '@woocommerce/base-context';
+import { getBlockTypes } from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
-import { useForcedLayout, getAllowedBlocks } from '../../../shared';
 
-export const Edit = ( { clientId }: { clientId: string } ): JSX.Element => {
+const excludedBlocks = [
+	'woocommerce/mini-cart',
+	'core/template-part',
+	'core/post-template',
+	'core/comment-template',
+];
+
+export const Edit = (): JSX.Element => {
 	const blockProps = useBlockProps();
-	const allowedBlocks = getAllowedBlocks( innerBlockAreas.EMPTY_MINI_CART );
 	const { currentView } = useEditorContext();
+	const allowedBlocks = [
+		...getBlockTypes()
+			.filter( ( block ) => {
+				if ( excludedBlocks.includes( block.name ) ) {
+					return false;
+				}
 
-	const defaultTemplate = ( [
-		[
-			'core/heading',
-			{
-				content: __(
-					'Empty mini cart content',
-					'woo-gutenberg-products-block'
-				),
-				level: 2,
-			},
-		],
-	].filter( Boolean ) as unknown ) as TemplateArray;
+				if ( block.parent && block.parent.length > 0 ) {
```

Comment:
I'm a newcomer, so sorry about that: but I'm not sure what's the intent here. Would you be so kind as to explain why blocks with parents are to be excluded? It would make me understand our intentions better 🙏 

Also, we could perhaps replace this with:

```
if ( block.parent?.length )
```

Reply:
Initially, I want to remove the blocks that have a parent from the allowed blocks to limit unwanted theme components like the previous post link. But doing so also removes other blocks that actually make sense to be included in the block list. I remove it for now and we should find another way to exclude blocks.